realm/realm-library/src/androidTest/java/io/realm/TestHelper.java

```diff
@@ -636,6 +637,36 @@ public static void populateForMultiSort(DynamicRealm realm) {
         realm.commitTransaction();
     }
 
+
+    /*
+     * Fields order test for Chained or Multi-Arguments Distinct()
+     *
+     * The idea is to interweave different values in 2's multiplier and 3's multiplier in a way that
+     * the outcome is different if the order of distinct* operations alternates. More numbers of
+     * fields can be constructed with the combination of multipliers in prime numbers such as 2, 3,
+     * and 5.
+     *
+     * An example is illustrated below.
+     *
+     * Object        : O1| O2| O3| O4| O5| O6
+     * indexedString : A | A | B | B | A | A
+     * indexedLong   : 1 | 1 | 1 | 2 | 2 | 2
+     *
+     * @param realm a {@link Realm} instance.
+     * @param numberOfBlocks number of times set of unique objects should be created.
+     */
+    public static void populateForFieldsOrderSwitch(Realm realm, long numberOfBlocks) {
```

Comment:
I would probably rename this to `populateForDistinctTests()` to increase readability. Right now it is a bit hard to decipher at a glance.